e2e/authorization_test.go

```diff
@@ -0,0 +1,164 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/GoogleCloudPlatform/prometheus-engine/e2e/kubeutil"
+	"github.com/GoogleCloudPlatform/prometheus-engine/e2e/operatorutil"
+	"github.com/GoogleCloudPlatform/prometheus-engine/pkg/operator"
+	monitoringv1 "github.com/GoogleCloudPlatform/prometheus-engine/pkg/operator/apis/monitoring/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func TestTLS(t *testing.T) {
+	t.Parallel()
+	tctx := newOperatorContext(t)
+	ctx := context.Background()
+
+	tctx.createOperatorConfigFrom(ctx, monitoringv1.OperatorConfig{
+		Features: monitoringv1.OperatorFeatures{
+			TargetStatus: monitoringv1.TargetStatusSpec{
+				Enabled: true,
+			},
+		},
+	})
+
+	c := tctx.Client()
+	const appName = "tls-insecure"
+	deployment, err := operatorutil.SyntheticAppDeploy(ctx, c, tctx.namespace, appName, []string{
+		"--tls-create-self-signed=true",
+	})
+	if err != nil {
+		tctx.Fatal(err)
+	}
+	if err := kubeutil.WaitForDeploymentReady(ctx, c, tctx.namespace, appName); err != nil {
+		kubeutil.DeploymentDebug(tctx.T, ctx, tctx.RestConfig(), tctx.Client(), tctx.namespace, appName)
+		tctx.Fatalf("failed to start app: %s", err)
+	}
+
+	tctx.Run("no-tls", tctx.subtest(func(ctx context.Context, t *OperatorContext) {
+		t.Parallel()
+
+		pm := &monitoringv1.PodMonitoring{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "collector-no-tls",
+				Namespace: t.namespace,
+			},
+			Spec: monitoringv1.PodMonitoringSpec{
+				Selector: metav1.LabelSelector{
+					MatchLabels: deployment.Spec.Template.Labels,
+				},
+				Endpoints: []monitoringv1.ScrapeEndpoint{
+					{
+						Port:     intstr.FromString(operatorutil.SyntheticAppPortName),
+						Scheme:   "https",
+						Interval: "5s",
+					},
+				},
+			},
+		}
+		if err := t.Client().Create(ctx, pm); err != nil {
+			t.Fatalf("create collector PodMonitoring: %s", err)
+		}
+
+		if err := operatorutil.WaitForPodMonitoringReady(ctx, t.Client(), pm, true); err != nil {
+			kubeutil.DaemonSetDebug(tctx.T, ctx, tctx.RestConfig(), tctx.Client(), tctx.namespace, operator.NameCollector)
+			t.Fatalf("collector not ready: %s", err)
+		}
+
+		var err error
+		if pollErr := wait.Poll(5*time.Second, 3*time.Minute, func() (bool, error) {
+			if err = t.Client().Get(ctx, client.ObjectKeyFromObject(pm), pm); err != nil {
+				return false, nil
+			}
+
+			const expected = "tls: failed to verify certificate: x509: certificate signed by unknown authority"
+			err = operatorutil.IsPodMonitoringScrapeEndpointFailure(pm, operatorutil.SyntheticAppPortName, func(message string) error {
+				if !strings.HasSuffix(message, expected) {
+					return fmt.Errorf("expected %q", expected)
```

Comment:
should we print what we got?

Reply:
and pass it through and fatal with proper error perhaps (instead of not passing the error from `IsPodMonitoringScrapeEndpointFailure` anywhere)

Reply:
Both are already happening:
1. The method wraps the error in the closure.
2. The error is logged outside of the polling.